src/erasure-code/ErasureCode.h

```diff
@@ -115,6 +115,13 @@ namespace ceph {
     int decode_concat(const std::map<int, bufferlist> &chunks,
 			      bufferlist *decoded) override;
 
+    static bool is_supported_profile(std::map<std::string, std::string> &profile_map) {
+      std::string plugin = profile_map["plugin"];
```

Comment:
is_supported_profile should take profile_map as const.  That will prevent using operator[], which is good as we don't want this method to be able to add empty strings for "plugin" or "technique".  Instead, check and reject any profiles that lack those two keys.

Reply:
This function now encodes an important bit of project policy, and needs a comment explaining that:

```
/**
 * is_supported_profile
 *
 * We only test a limited subset of implemented plugin/technique combinations.  Combinations for
 * which this function returns false will be rejected unless the "experimental_ec_profiles" experimental
 * feature is enabled.  See the "osd erasure-code-profile set" handler in
 * OSDMonitor::prepare_command_impl.  As new plugin/techique combinations are declared stable,
 * this function should be updated.
 */
```

Reply:
Fixed